File: BE/Common/src/main/java/com/pbl5/common/entity/product/ProductVariant.java
package com.pbl5.common.entity.product;

import com.pbl5.common.entity.IdBaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "product_variants")
@Getter
@Setter
public class ProductVariant extends IdBaseEntity {

    @Column(length = 50, name = "`key`")
    private String key;
    @Column(length = 50)
    private String value;
    private Integer quantity;
    @Column(length = 64)
    private String photo;

    @Column(name = "parent_id")
    private Integer parentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Product product;

    @Override
    public String toString() {
        return "ProductVariant{" +
                "key='" + key + '\'' +
                ", value='" + value + '\'' +
                ", quantity=" + quantity +
                ", photo='" + photo + '\'' +
                '}';
    }
}

File: FE/FE_Admin/src/models/ProductVariant.ts
// Cấu trúc biến thể
interface ProductVariant {
    id: number;
    key: string;    // Ví dụ: "Màu sắc", "Kích cỡ"
    value: string;  // Ví dụ: "Đỏ", "S"
    quantity: number;
    photo?: string;
    parentId: number | null;
}

// Cấu trúc nhóm biến thể
interface ProductVariantGroup {
    name: string;
    variants: ProductVariant[];
}